### PR TITLE
feat(condo): DOMA-10067 basic unitName normalization

### PR DESCRIPTION
**Pull request by YEgorLu**
Made normalizeUnitName function in accordance with what was agreed upon

Placed it in billing domain because at first it will be used here, maybe move it somewhere when making plugin for unitNameKey

Didn't find good slugify package, others does not work with \ plan to remove CommonJS imports. One package, that we could use, has difficulties with shared charMap state. They have instruction on how to reset it's state, but it isn't working for me. So i copied package and added methods that allow resetting state and extending local-based state. These methods are needed in current task and allow to reuse package. I need different charMaps for different locales to address collisions between translations of symbols in different locales.

### CODE REVIEW

File: packages/slugify/slugify.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+declare const slugify: {
```
Comment (SavelevMatthew) [inline packages/slugify/slugify.d.ts:1]: This package must be written in Typescript, and then built properly if you want to publish it

Comment (SavelevMatthew) [reply packages/slugify/slugify.d.ts:1]: But I'm not sure we need this one

File: packages/slugify/slugify.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+const _charMap = require('@open-condo/slugify/config/charMap')
+const _locales = require('@open-condo/slugify/config/locales')
+    
+
+;(function (name, root, factory) {
```
Comment (SavelevMatthew) [inline packages/slugify/slugify.js:5]: ;(

Comment (YEgorLu) [reply packages/slugify/slugify.js]: removed

File: packages/slugify/slugify.d.ts (hunk 1 above)
Comment (YEgorLu) [reply packages/slugify/slugify.d.ts]: removed

File: apps/condo/domains/billing/utils/unitName.utils.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    replacement: DELIMITER_IN_RESULT,
+    locale,
+    strict: true,
+    lower: true,
+})
+
+const normalizeUnitName = (unitName, locale = conf.DEFAULT_LOCALE) => {
+    locale = locale.toLowerCase()
+
+    const steps = [
```
Comment (pahaz) [inline apps/condo/domains/billing/utils/unitName.utils.js:35]: looks like a const

File: apps/condo/domains/billing/utils/unitName.utils.spec.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
+const  { normalizeUnitName } = require('./unitName.utils')
+
+const TEST_CASES = {
```
Comment (pahaz) [inline apps/condo/domains/billing/utils/unitName.utils.spec.js:3]: As for me it looks like we can remove all `-` in results (optional)

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/billing/utils/unitName.utils.js (hunk 3 above)
Comment (YEgorLu) [reply apps/condo/domains/billing/utils/unitName.utils.js]: moved to global const

File: apps/condo/domains/billing/utils/unitName.utils.spec.js (hunk 4 above)
Comment (YEgorLu) [reply apps/condo/domains/billing/utils/unitName.utils.spec.js:3]: remove delimiter? can not understand why. If you mean normalizeUnitName fn result, some delimiter required to see, where is a word, where is a separate symbol

**Comment by YEgorLu** [comment]
kept npm slugify package, removed dictionaries, changed one regex for sonarcloud

This solves one bug. Some more research required to solve problem more globally

**Review by dkoviazin — APPROVED** [review]

**Pull request merged**